util/util_tls.go

```diff
@@ -76,20 +76,42 @@ func CreateSelfCert(
 			CommonName:   host,
 		},
 
-		IPAddresses: []net.IP{net.ParseIP(host)},
-		DNSNames:    []string{"localhost"},
-
 		NotBefore:          time.Now(),
 		NotAfter:           time.Now().AddDate(1, 0, 0),
 		SignatureAlgorithm: x509.SHA256WithRSA,
 
+		IsCA: true,
 		KeyUsage: x509.KeyUsageKeyEncipherment |
-			x509.KeyUsageDigitalSignature,
+			x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
 
 		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth},
 		BasicConstraintsValid: true,
 	}
 
+	tmpl.IPAddresses = append(tmpl.IPAddresses, net.ParseIP("127.0.0.1"))
+
+	if ip := net.ParseIP(host); ip != nil {
+		tmpl.IPAddresses = append(tmpl.IPAddresses, ip)
+	} else {
+		ips, err := net.LookupIP(host)
+		if err != nil {
+			ctx.WithFields(log.Fields{
```

Comment:
Hi @vladimirvivien,

Could you possibly use the logger's `WithField` and `WithError` functions?

```go
ctx.WithError(err).WithField("host", host).Debug("failed to lookup IP for host")
```